### PR TITLE
RYA-450 Implemented a Kafka backed QueryChangeLogSource.

### PR DESCRIPTION
**Pull request by kchilton2**
## Description
Implemented the KafkaQueryChangeLogSource.

### Tests
Added new tests.

### Links
https://issues.apache.org/jira/browse/RYA-450

### CODE REVIEW

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java [hunk 1]
```diff
@@ -48,7 +51,28 @@
...
+        return ryaInstance + QUERY_CHANGE_LOG_TOPIC_SUFFIX;
+    }
+
+    /**
+     * Get the Rya instance name from a Kafka topic name that has been used for a {@link QueryChnageLog}.
+     * <p/>
+     * This is the inverse function of {@link #queryChangeLogTopic(String)}.
+     *
+     * @param changeLogTopic - The topic to evaluate. (not null)
+     * @return If the topic is well formatted, then the Rya instance name that was part of the topic name.
```
Comment (isper3at) [inline extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java:63]: The Rya instance name if the topic is well formatted

File: extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/kafka/KafkaQueryChangeLogSource.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    /**
+     * Used by the service to configure how often it polls the Kafka Server for topics.
+     */
+    private final Scheduler scheduler;
+
+    /**
+     * Which Kafka Server this source represents.
+     */
+    private final String kafkaBootstrapServers;
```
Comment (isper3at) [inline extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/kafka/KafkaQueryChangeLogSource.java]: doc says server, the name says Servers.

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java (hunk 1 above)
Comment (kchilton2) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java:63]: I'm not sure what you're asking for with this comment. You want me to just reverse the order of the sentence? I'm not changing that.

File: extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/kafka/KafkaQueryChangeLogSource.java (hunk 2 above)
Comment (kchilton2) [reply extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/kafka/KafkaQueryChangeLogSource.java]: done

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java (hunk 1 above)
Comment (isper3at) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaTopics.java:63]: it just reads weird to me

**Pull request merged**